src/odoc/html_page.ml

```diff
@@ -54,4 +58,46 @@ let extra_documents args unit ~syntax =
       []
   | None, None -> []
 
+exception Missing_asset of string
+
+let asset_documents parent_id children asset_paths =
+  let asset_names =
+    List.filter_map
+      (function Lang.Page.Asset_child name -> Some name | _ -> None)
+      children
+  in
+  let rec extract paths name =
+    match paths with
+    | [] -> raise (Missing_asset name)
+    | x :: xs when Fpath.basename x = name -> (xs, (name, x))
+    | x :: xs ->
+        let rest, elt = extract xs name in
+        (x :: rest, elt)
+  in
+  match List.fold_left_map extract asset_paths asset_names with
+  | unmatched, paired ->
+      List.iter
+        (fun asset ->
+          Error.raise_warning
+            (Error.filename_only "this asset was not declared as a child of %s"
+               (Paths.Identifier.name parent_id)
+               (Fs.File.to_string asset)))
+        unmatched;
+      List.map
+        (fun (name, path) ->
+          let asset_id = Paths.Identifier.Mk.asset_file (parent_id, name) in
+          let url = Odoc_document.Url.Path.from_identifier asset_id in
+          Odoc_document.Types.Document.Asset { url; src = path })
+        paired
+  | exception Missing_asset name ->
```

Comment:
Handling the exception here is a bit strange. Only a warning is issued, but this stops the process of assets whenever one is not found...